### PR TITLE
Fix puppet parser on Foreman > 1.6 again

### PR DESCRIPTION
**Pull request by ares**
Also fixes case where one fact is upcase, second downcase

### CODE REVIEW

File: lib/foreman_discovery/engine.rb [hunk 1]
```diff
@@ -73,9 +73,16 @@ class Engine < ::Rails::Engine
     end
     # Include extensions to models in this config.to_prepare block
     config.to_prepare do
+
+      begin
+        require_dependency 'puppet_fact_parser'
+        ::PuppetFactParser.send :include, PuppetFactParserExtensions
+      rescue LoadError
+        Rails.logger.warn 'PuppetFactParser not found, ignoring the fix for that'
```
Comment (GregSutcliffe) [inline lib/foreman_discovery/engine.rb]: nitpick: Its not really a fix, it's an extension. Mind if I reword that as "not found, not loading Parser extensions" ?

Comment (ares) [inline lib/foreman_discovery/engine.rb]: please do :-)

**Comment by ares** [comment]
I did the reword and also fix one more issue (when fact is not present)

**Comment by lzap** [comment]
How do I reproduce this one? Develop works fine against discovery develop for me.

**Comment by GregSutcliffe** [comment]
@lzap check the PuppetFactParser extensions are getting loaded (maybe at some debug to check the chained methods are being called at import time)

**Comment by lzap** [comment]
I can confirm that, looks like we can merge and release then!

**Comment by lzap** [comment]
Merged as 213eedd for Discovery 1.4, thanks @ares !

**Pull request merged**